src/utils/xwayland/satellite.rs

```diff
@@ -307,6 +322,8 @@ fn spawn_and_wait(
         }
     };
 
+    let _ = to_main.send(ToMain::Pid(Some(child.id())));
```

Comment:
Another possible way to do it without PIDs is to create a separate Wayland socket for our xwayland-satellite, similarly to how we create a separate Wayland connection for sandboxed clients. Not sure how much effort that would be.